examples/src/tower/server.rs

```diff
@@ -71,13 +72,9 @@ where
     }
 
     fn call(&mut self, req: HyperRequest<Body>) -> Self::Future {
-        let mut svc = self.inner.clone();
+        // Do async work here....
 
-        Box::pin(async move {
-            // Do async work here....
-
-            svc.call(req).await
-        })
+        Box::pin(self.inner.call(req).err_into::<Self::Error>())
```

Comment:
```suggestion
        let clone = self.inner.clone();
        let mut inner = std::mem::replace(&mut self.inner, clone);
        Box::pin(async move {
            // Do extra async work here...
            inner.call(req).await.map_err::<Self::Error>()
        })
```

This was we still make it clear to users where to add additional async code.